tt_metal/jit_build/build.cpp

```diff
@@ -558,6 +559,18 @@ void JitBuildState::weaken(const string& log_file, const string& out_dir) const
     }
 }
 
+void JitBuildState::extract_zone_src_locations(const string& log_file) const
+{
+    static std::atomic<bool> new_log = true;
+    if (tt::tt_metal::getDeviceProfilerState()) {
+        if (new_log.exchange(false) && std::filesystem::exists(tt::tt_metal::PROFILER_ZONE_SRC_LOCATIONS_LOG)) {
+            std::remove(tt::tt_metal::PROFILER_ZONE_SRC_LOCATIONS_LOG.c_str());
+        }
+        string cmd = "cat " +  log_file + " | grep KERNEL_PROFILER";
```

Comment:
why "cat file | grep KERNEL_PROFILER" rather than "grep KERNEL_PROFILER logfile"?

Reply:
Will explain in code comments